zerver/fixtures/markdown_test_cases.json

```diff
@@ -251,6 +251,36 @@
       "expected_output": "<p>1. A<br>\n3. B<br>\n 2. C<br>\n1. D</p>",
       "text_content": "1. A\n3. B\n 2. C\n1. D"
     },
+    {
+      "name": "auto_renumbered_list_paragraphs",
+      "input": "1. A\n1. B\n\n1. A\n1. B",
+      "expected_output": "<p>1. A<br>\n2. B</p>\n<p>1. A<br>\n2. B</p>",
+      "text_content": "1. A\n2. B\n1. A\n2. B"
+    },
```

Comment:
I'm not sure this is the right answer for how numbered lists should work in Zulip.  I think it's probably a lot more common for folks to have a blank line in between some of their bullets than to have two separate auto-numbered ordered lists in the document.

So I definitely want this case to just be 1, 2, 3, 4.  I can imagine a case where there's a sentence or something in between the numbers wanting it to be 1, 2, 1, 2, but I think I'd be happy with it being 1, 2, 3, 4; whichever is easier to implement is fine.

@rishig FYI in case you have thoughts.

Reply:
yeah, I think in every case (where it is easy to do so) we want 1 1 1 1 to convert to 1 2 3 4, even if there are paragraph breaks. If someone intends 1 2 1 2 I think they won't be sad to have to write those numbers explicitly.